pkg/port/iscsit/session.go

```diff
@@ -286,6 +286,22 @@ func (tq *taskQueue) Pop() interface{} {
 	return item
 }
 
+func (tq *taskQueue) Delete(x interface{}) {
+	item := x.(*iscsiTask)
+	old := *tq
+	n := len(old)
+	for i, t := range old {
+		if t == item {
+			if i == 0 && n == 1 {
+				tq = nil
+				break
+			}
+			*tq = append(old[:i], old[i+1:]...)
```

Comment:
Does this mean the item memory is free to be garbage collected after this line?

Reply:
yes

Reply:
We seem to walk this list twice in the same function  -- first to get a reference to the task (iscsid.go:828), then we process it after which we go searching for it again, in effect making it n^2. Perhaps it's out of scope here, an avl tree or something would be better? In any caseaA function that pops the iscsiTask would prevent that and would also prevent the leak that is still there on iscsid.go:847.

For this PR, @payes are you sure we are not leaking on line 847?